### PR TITLE
STM32 EEPROM Emulation

### PR DESCRIPTION
**Pull request by yiancar**
- Added EEPROM emulation libaries from libmaple and Arduino_STM32. https://github.com/rogerclarkmelbourne/Arduino_STM32 and https://github.com/leaflabs/libmaple.
- Renamed teensy EEPROM library and added conditional selection of library.
- Remapped EEPROM memory map for 16 byte blocks (as is with STM32f3xx MCUs).
- Added EEPROM initialization in main.c of Chibios.
- Added EEPROM format to clear the emulated pages when EEPROM is marked as invalid.

### CODE REVIEW

File: tmk_core/common/eeconfig.c [hunk 1]
```diff
@@ -2,13 +2,21 @@
 #include <stdbool.h>
 #include "eeprom.h"
 #include "eeconfig.h"
+#include "hal.h"
```
Comment (drashna) [inline tmk_core/common/eeconfig.c]: Looks like this is causing issues.

Comment (yiancar) [reply tmk_core/common/eeconfig.c]: uups
Sorry too less coffee:)

**Comment by yiancar** [comment]
@jackhumbert did you try this? any progress to merge?

**Comment by jackhumbert** [comment]
Perfect! Thanks for the quick work on this :)

**Pull request merged**